nav2_bt_navigator/behavior_trees/navigate_w_replanning_and_recovery.xml

```diff
@@ -5,21 +5,31 @@
 <root main_tree_to_execute="MainTree">
   <BehaviorTree ID="MainTree">
     <RecoveryNode number_of_retries="6" name="NavigateRecovery">
-      <Sequence name="NavigateWithReplanning">
-        <RateController hz="1.0">
-          <Fallback name="GoalReached">
-            <GoalReached/>
-            <RecoveryNode number_of_retries="1" name="ComputePathToPose">
-              <ComputePathToPose goal="{goal}" path="{path}" planner_property="GridBased"/>
-              <ClearEntireCostmap service_name="global_costmap/clear_entirely_global_costmap"/>
-            </RecoveryNode>
-          </Fallback>
-        </RateController>
-        <RecoveryNode number_of_retries="1" name="FollowPath">
-          <FollowPath path="{path}"/>
-          <ClearEntireCostmap service_name="local_costmap/clear_entirely_local_costmap"/>
-        </RecoveryNode>
-      </Sequence>
+      <FirstCompletion name="NavigateWithReplanning">
```

Comment:
Comment (don't think there's anything to be done about it here): These Behavior trees are really starting to look nasty trying to follow. Is there a way to simplify these more or integrate sub-trees to try to make them more approachable.

Reply:
We should be able to break out sub-trees into separate files. That's one of the features of V3 of the behavior tree library. We haven't tackled that yet.

I'd imaging a subtree file for each of the planning and control branches, and maybe recoveries if they get complicated enough.